opennms-config-model/src/main/java/org/opennms/netmgt/config/notifications/Notification.java

```diff
@@ -28,236 +28,213 @@
 
 package org.opennms.netmgt.config.notifications;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import org.opennms.features.config.util.json.YesNoDeserializer;
+import org.opennms.netmgt.config.utils.ConfigUtils;
+
 import java.io.Serializable;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
 import java.util.Optional;
 
-import javax.xml.bind.annotation.XmlAccessType;
-import javax.xml.bind.annotation.XmlAccessorType;
-import javax.xml.bind.annotation.XmlAttribute;
-import javax.xml.bind.annotation.XmlElement;
-import javax.xml.bind.annotation.XmlRootElement;
-import javax.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
-
-import org.opennms.core.xml.ValidateUsing;
-import org.opennms.core.xml.YesNoAdapter;
-import org.opennms.netmgt.config.utils.ConfigUtils;
-
-@XmlRootElement(name = "notification")
-@XmlAccessorType(XmlAccessType.FIELD)
-@ValidateUsing("notifications.xsd")
 public class Notification implements Serializable {
     private static final long serialVersionUID = 2L;
 
-    @XmlAttribute(name = "name", required = true)
-    private String m_name;
-
-    @XmlAttribute(name = "status", required = true)
-    private String m_status;
+    private String name;
 
-    @XmlAttribute(name = "writeable")
-    @XmlJavaTypeAdapter(YesNoAdapter.class)
-    private Boolean m_writeable;
+    private String status;
 
-    @XmlElement(name = "uei", required = true)
-    private String m_uei;
+    @JsonDeserialize(using = YesNoDeserializer.class)
+    private Boolean writeable;
 
-    @XmlElement(name = "description")
-    private String m_description;
+    private String uei;
 
-    @XmlElement(name = "rule", required = true)
-    private Rule m_rule;
+    private String description;
 
-    @XmlElement(name = "notice-queue")
-    private String m_noticeQueue;
+    private Rule rule;
 
-    @XmlElement(name = "destinationPath", required = true)
-    private String m_destinationPath;
+    private String noticeQueue;
 
-    @XmlElement(name = "text-message", required = true)
-    private String m_textMessage;
+    @JsonProperty("destinationPath")
```

Comment:
do we need the annotation for a json fieldwith the same name as the Java attribute?

Reply:
Because field in xml are mostly kebab-case. The object mapper is set to convert to kebab-case. For destinationPath it is camelCase. Unless we change the attribute case in xsd otherwise we need that hardcode name.

Reply:
got it!